plugins/coverbrowser.koplugin/bookinfomanager.lua

```diff
@@ -617,25 +617,27 @@ end
 
 -- Background extraction management
 function BookInfoManager:collectSubprocesses()
+    self.subprocesses_collector = nil
+
     -- We need to regularly watch if a sub-process has terminated by
     -- calling waitpid() so this process does not become a zombie hanging
     -- around till we exit.
     if #self.subprocesses_pids > 0 then
-        local i = 1
-        while i <= #self.subprocesses_pids do -- clean in-place
+        -- In-place removal, hence the reverse iteration.
+        for i = #self.subprocesses_pids, 1, -1 do
             local pid = self.subprocesses_pids[i]
             if FFIUtil.isSubProcessDone(pid) then
                 table.remove(self.subprocesses_pids, i)
                 -- Prevent has been issued for each bg task spawn, we must allow for each death too.
                 UIManager:allowStandby()
-            else
-                i = i + 1
             end
```

Comment:
> Can't table.remove a table you're currently iterating on in order :s.

I don't think there was any issue: it's not like we were using ipairs, which would be messy with table.remove. Here, we were just iterating in ascending order by using an index and keepint it unchanged when table.remove'ing it.

I'm fine with your change, which makes it more readable - but I really think it fixes nothing.

Reply:
True ;o).